src/context/authContext.tsx

```diff
@@ -62,15 +62,19 @@ export const AuthProvider = ({ children }: { children: ReactNode }) => {
                 }),
             })
 
-            if (response.ok) {
-                const updatedUserData: any = await response.json()
-                if (updatedUserData.success) {
-                    fetchUser()
-                }
-            } else {
-                throw new Error(response.statusText)
+            if (response.status === 409) {
+                toast.close(toastIdRef.current)
+                toastIdRef.current = toast({
+                    status: 'error',
+                    title: 'Username already exists',
+                }) as ToastId
+
```

Comment:
Nit: same as before, i think we should be able to just reuse the message from the api instead of duplicating it in frontend